### PR TITLE
Fix #12 by adding a _.mergeWith customizer function

### LINKED ISSUE #12: Lists are handled unintuitively
```yaml
# first.yml
hello:
- world
- alex

# second.yml
hello:
- moritz

# yaml-merge first.yml second.yml
hello:
- moritz
- alex
```

It makes sense, as `.hello[0]` in `first.yml` gets overwritten with `.hello[0]` from `second.yml`, but when using this to merge e.g. configuration files, most people probably expect something like `["world", "alex", "moritz"]` from that. If someone really wants the second behaviour, you could still allow the following YAML file to override values:

```yaml
hello:
  0: moritz
```

To implement this, there's https://lodash.com/docs/4.17.15#mergeWith:
```javascript
_.mergeWith({}, ...files, (objValue, srcValue, key, object, source, stack) => Array.isArray(objValue) && Array.isArray(srcValue) ? [...objValue, ...srcValue] : undefined)
```

### PR DESCRIPTION
**Pull request by moqmar**
See #12 for details

### CODE REVIEW

**Comment by moqmar** [comment]
`0: moritz` will still override existing entries in this approach because it results in `{ "0": "moritz" }`, and `Array.isArray` correctly returns `false` for that construct.

**Comment by alexlafroscia** [comment]
Thanks for the PR! I think you're right in the fact that it makes more sense that arrays are fully merged, rather than some items being replaced the way they are right now.

This _would_ constitute a breaking change in the library, but I am personally OK with that.

File: index.js [hunk 1]
```diff
@@ -20,7 +20,10 @@ function readAsJSON(fileName) {
 function yamlMerge(...from) {
   const files = from.map((path) => readAsJSON(path));
 
-  const outputJSON = _.merge({}, ...files);
+  const outputJSON = _.mergeWith({}, ...files, (objValue, srcValue, key, object, source, stack) => {
+    if (Array.isArray(objValue) && Array.isArray(srcValue)) return [...objValue, ...srcValue];
+    return undefined; // handle it just like with _.merge
```
Comment (alexlafroscia) [inline index.js:25]: Just to make sure I understand, returning `undefined` means "do whatever `lodash` would normally do by default" and not "replace with `undefined`", right?

**Comment by alexlafroscia** [comment]
Since we're changing some behavior here, I would really appreciate it if you could add a test case! That would help make sure that it's really clear how this library merges arrays.

File: index.js (hunk 1 above)
Comment (moqmar) [reply index.js:25]: Yes, exactly, that's according to their specification.

**Comment by moqmar** [comment]
I'll try to do this in the next couple of days.

**Comment by thewizarodofoz** [comment]
Hey @alexlafroscia @moqmar any updates on this? We would really love to see thing merged as we are blocked by currenty behaviour

**Comment by alexlafroscia** [comment]
Not from me -- still waiting for some tests!

**Comment by moqmar** [comment]
Whoops, totally forgot about this - I've added a test case for this now. :upside_down_face:

**Review by alexlafroscia — APPROVED** [review]

**Comment by alexlafroscia** [comment]
Thanks @moqmar !

**Comment by alexlafroscia** [comment]
`v3.0.0` released with this change 🎉

File: index.js (hunk 1 above)
Comment (greenimpala) [reply index.js:25]: This now breaks my use cases so I am sticking to 2.0. Would it have been better to provide a customizer fn as a prop for the node package for custom use cases such as these.

**Pull request merged**